File: Android/app/src/main/java/com/scurab/android/anuitorsample/SampleApplication.java
package com.scurab.android.anuitorsample;

import android.widget.Button;

import androidx.fragment.app.Fragment;
import androidx.multidex.MultiDexApplication;

import com.scurab.android.anuitor.extract2.BaseExtractor;
import com.scurab.android.anuitor.extract2.DetailExtractor;
import com.scurab.android.anuitor.extract2.ExtractingContext;
import com.scurab.android.anuitor.hierarchy.IdsHelper;
import com.scurab.android.anuitor.service.AnUitorClientConfig;
import com.scurab.android.anuitor.service.AnUitorService;
import com.scurab.android.anuitorsample.common.BaseFragment;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Created by jbruchanov@gmail.com
 *
 * @since 2014-05-15.
 */
public class SampleApplication extends MultiDexApplication {

    @Override
    public void onCreate() {
        super.onCreate();

        AnUitorClientConfig.addTypeHighlighting(Button.class, "rgba(255, 0, 255, 0.15)");
        AnUitorClientConfig.addPropertyHighlighting("text.*|.*TextColor.*|CanResolveText.*", "rgba(255, 0, 255, 1)");
        AnUitorClientConfig.addPointerIgnoreViewId(R.id.pointer_ignore);
        AnUitorService.startService(this, 8081, 0, true, null);
        IdsHelper.loadValues(R.class);

        DetailExtractor.registerRenderArea(DrawOutsideBoundsFragment.HelpTextView.class,
                (view, outRect) -> view.getDrawingSize(outRect));

        DetailExtractor.registerExtractor(BaseFragment.class, new BaseExtractor() {

            @Nullable
            @Override
            public Class<?> getParent() {
                return Fragment.class;
            }

            @Override
            protected void onFillValues(@NotNull Object item, @NotNull ExtractingContext context) {
                BaseFragment baseFragment = (BaseFragment) item;
                context.put("FakePresenter", baseFragment.getFakePresenter());
            }
        });
    }
}

File: Android/app/src/main/java/com/scurab/android/anuitorsample/SampleApplication.kt
package com.scurab.android.anuitorsample

import android.view.View
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.multidex.MultiDexApplication
import com.scurab.android.uitor.Constants
import com.scurab.android.uitor.extract.RenderAreaWrapper
import com.scurab.android.uitor.extract2.*
import com.scurab.android.uitor.hierarchy.IdsHelper
import com.scurab.android.uitor.service.UitorClientConfig
import com.scurab.android.uitor.service.UitorService
import com.scurab.android.anuitorsample.common.BaseFragment

class SampleApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()

        UitorClientConfig.addTypeHighlighting(Button::class.java, "rgba(255, 0, 255, 0.5)")
        UitorClientConfig.addPropertyHighlighting("text.*|.*TextColor.*|CanResolveText.*", "rgba(255, 0, 255, 1)")
        UitorClientConfig.addPointerIgnoreViewId(R.id.pointer_ignore)
        UitorClientConfig.setResourcesInSnapshot(true)
        UitorService.startService(this, 8081)
        IdsHelper.loadValues(R::class.java)

        DetailExtractor.registerRenderArea(DrawOutsideBoundsFragment.HelpTextView::class.java,
                RenderAreaWrapper { view, outRect -> view.getDrawingSize(outRect) })

        DetailExtractor.registerExtractor(BaseFragment::class.java, object : BaseExtractor() {
            override val parent: Class<*> = Fragment::class.java

            override fun onFillValues(item: Any, context: ExtractingContext) {
                val baseFragment = item as BaseFragment
                context.put("FakePresenter", baseFragment.fakePresenter)
            }
        })

        //replace the ViewExtractor with custom one
        DetailExtractor.registerExtractor(View::class.java, object : ViewExtractor() {

            override fun onFillValues(item: Any, context: ExtractingContext) {
                super.onFillValues(item, context)
                context.data[Constants.OWNER]
                        ?.let { (it as? IFragmentDelegate)?.fragment as? BaseFragment }
                        ?.let {
                            context.put("FakePresenter", it.fakePresenter)
                        }
            }
        })
    }
}